### PR TITLE
refactor: continue with refactor and commenting service selection

### PR DESCRIPTION
**Pull request by ivanahuckova**
Follow up on https://github.com/grafana/explore-logs/pull/266, this PR:
- removes live tailing logic (hopefully the last :D)
- adds couple more comments
- refactors the code -> moving some logic into specific function (`orderResults`)
- removes unused variables
- fixes tiny bug where GrotError to enable volume endpoint was shown if there were no `servicesToQuery`, but we want to check `servicesByVolume`. This is because `servicesToQuery` can be empty whenever you search for service that does not exist. And `servicesByVolume` has info about all services.

### CODE REVIEW

File: src/Components/Index/ServiceSelectionComponent.tsx [hunk 1]
```diff
@@ -172,38 +173,12 @@ export class ServiceSelectionComponent extends SceneObjectBase<ServiceSelectionC
...
-                          return 1;
-                        } else {
-                          return (b.fields[1].state?.calcs?.max || 0) - (a.fields[1].state?.calcs?.max || 0);
-                        }
-                      });
-                    })
-                  );
-                },
+                () => (source: Observable<DataFrame[]>) =>
+                  source.pipe(map((data: DataFrame[]) => orderResults(data, getFavoriteServicesFromStorage(ds)))),
```
Comment (matyax) [inline src/Components/Index/ServiceSelectionComponent.tsx:181]: <3

File: src/services/store.ts [hunk 2]
```diff
@@ -2,11 +2,12 @@ import pluginJson from '../plugin.json';
 
 const SERVICES_LOCALSTORAGE_KEY = `${pluginJson.id}.services.favorite`;
 
-export function getFavoriteServicesFromStorage(ds: any): string[] {
-  if (!ds || typeof ds !== 'string') {
+// This should be a string, but we'll accept anything and return an empty array if it's not a string
```
Comment (matyax) [inline src/services/store.ts:5]: Why? Shouldn't it be string or null?

Comment (ivanahuckova) [reply src/services/store.ts:5]: It can be `VariableValue | null | undefined` and `VariableValue` can be array of basically anything.

Comment (matyax) [reply src/services/store.ts:5]: Oh, is coming from there :(

**Review by matyax — APPROVED** [review]
💯+💯=2*💯

**Pull request merged**